### PR TITLE
Added changelog for v0.12

### PR DESCRIPTION
**Pull request by Razican**
This is a preliminary changelog for the Boa v0.12 version. Feel free to suggest improvements.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,57 @@
 # CHANGELOG
 
-# 0.11.0 (2021-01-14)
+# [0.12.0 (YYYY-MM-DD) - `Set`, accessors, `@@toStringTag` and panic reduction](https://github.com/boa-dev/boa/compare/v0.11.0...v0.12.0)
```
Comment (0x7D2B) [inline CHANGELOG.md:3]: didn't panics increase :(

Comment (Razican) [reply CHANGELOG.md:3]: Good point. So, many of the big panics were resolved, but some internal changes introduced some extra panics. Nevertheless, we still have a couple of changes in the pipeline that would reduce the panics.

Comment (0x7D2B) [reply CHANGELOG.md:3]: new plan: fix enough panics to make that statement true

Comment (0x7D2B) [reply CHANGELOG.md:3]: panic reduction? how about panic elimination!

**Review by RageKnify — APPROVED** [review]

**Review by jasonwilliams — APPROVED** [review]

**Pull request merged**